config/configelasticsearch/esclient.go

```diff
@@ -147,10 +149,31 @@ func (cfg *ClientConfig) ToClient(
 		EnableDebugLogger: false, // TODO
 		Logger:            &esLogger,
 
-		Instrumentation: elasticsearch.NewOpenTelemetryInstrumentation(telemetry.TracerProvider, false),
+		Instrumentation: func() elastictransport.Instrumentation {
+			// only set tracing if enabled and not the noop tracer
+			if !isNoopTracer(telemetry.TracerProvider) {
+				return elasticsearch.NewOpenTelemetryInstrumentation(telemetry.TracerProvider, false)
+			}
+			return nil
+		}(),
 	})
 }
 
+func isNoopTracer(provider trace.TracerProvider) bool {
+	if provider == nil {
+		return true
+	}
+
+	// The actual implementation of the no-op tracer (tracer.noopNoContextTracer)
+	// is not exported by the OpenTelemetry collector, so we cannot directly check for it.
+	// Instead, we create a span and verify whether it records data.
+	//
+	// If the tracer is a no-op, the span will not record any events or attributes.
+	// We avoid ending the span to prevent unnecessary delivery.
+	_, span := provider.Tracer("es-config").Start(context.Background(), "noop-test")
```

Comment:
@dmathieu Are you aware of any mechanism to verify if the tracer provider is NO-OP without leaking this span?

cc @gregkalapos

Reply:
Another side effect of this is that if sampling is enabled, the span may not be recording, the tracer is enabled though.

Reply:
There's no way to retrieve this information cleanly. If you really want to do this, you could possibly print a `%#v` of the tracer, and check that it is a `&trace.tracer{`.
That won't work if folks are using a custom SDK though (such as [apmotel](https://github.com/elastic/apm-agent-go/tree/main/module/apmotel)).

In the future, this can be checked with the in-development [Enabled](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/api.md#enabled) method.
That specification is not yet enabled in the Go SDK though.

Reply:
Good point, maybe another option would be checking the span type which seems to be exported (`"go.opentelemetry.io/otel/trace/noop"`)

```suggestion
	_, span := provider.Tracer("es-config").Start(context.Background(), "noop-test")
	_, ok := span.(noop.Span)
	return ok

```

Reply:
That still won't work with custom SDKs.

Reply:
That could work, yes.

For the record, if a span is not sampled, a non-recording span is returned, but that's not a noop.
https://github.com/open-telemetry/opentelemetry-go/blob/007fb1f65bdd5df6e0ea851c7b05ca047bc48f4e/sdk/trace/tracer.go#L152

Reply:
> There's no way to retrieve this information cleanly. If you really want to do this, you could possibly print a %#v of the tracer, and check that it is a &trace.tracer{.
That won't work if folks are using a custom SDK though (such as [apmotel](https://github.com/elastic/apm-agent-go/tree/main/module/apmotel)).

I am not aware of the collector allowing custom SDKs through its configuration, and it looks like that if telemetry is not disabled it uses the Go SDK tracer provider which is an exported type https://github.com/open-telemetry/opentelemetry-collector/blob/v0.118.0/service/telemetry/tracer.go#L66 

Since the `Enabled()` option is not available yet, we could enable the instrumentation only if the sdk tracer is used: 
https://github.com/elastic/opentelemetry-lib/pull/149/commits/2b88a3dd90543159f95bfed79744428a0d6df3a7